alphadia/workflow/peptidecentric.py

```diff
@@ -319,121 +282,198 @@ def get_batch_plan(self):
+            fragments_df = pd.concat(fragments)
 
-    def check_epoch_conditions(self):
-        continue_calibration = False
+            self.reporter.log_string(
+                f"=== Step {current_step}, extracted {len(feature_df)} precursors and {len(fragment_df)} fragments ===",
+                verbosity="progress",
+            )
 
-        self.reporter.log_string(
-            "=== checking if epoch conditions were reached ===", verbosity="info"
-        )
-        if self.dia_data.has_ms1:
-            if self.com.ms1_error > self.config["search"]["target_ms1_tolerance"]:
-                self.reporter.log_string(
-                    f"❌ {'ms1_error':<15}: {self.com.ms1_error:.4f} > {self.config['search']['target_ms1_tolerance']}",
-                    verbosity="info",
-                )
-                continue_calibration = True
-            else:
-                self.reporter.log_string(
-                    f"✅ {'ms1_error':<15}: {self.com.ms1_error:.4f} <= {self.config['search']['target_ms1_tolerance']}",
-                    verbosity="info",
-                )
+            precursor_df = self.fdr_correction(
+                features_df, fragments_df, self.optimization_manager.classifier_version
+            )
 
-        if self.com.ms2_error > self.config["search"]["target_ms2_tolerance"]:
             self.reporter.log_string(
-                f"❌ {'ms2_error':<15}: {self.com.ms2_error:.4f} > {self.config['search']['target_ms2_tolerance']}",
+                f"=== FDR correction performed with classifier version {self.optimization_manager.classifier_version} ===",
                 verbosity="info",
             )
-            continue_calibration = True
-        else:
+
+            precursors_01FDR = len(precursor_df[precursor_df["qval"] < 0.01])
+
             self.reporter.log_string(
-                f"✅ {'ms2_error':<15}: {self.com.ms2_error:.4f} <= {self.config['search']['target_ms2_tolerance']}",
-                verbosity="info",
+                f"=== Checking if minimum number of precursors for optimization found yet; minimum number is {self.config['calibration']['optimization_lock_target']} ===",
+                verbosity="progress",
             )
 
-        if self.com.rt_error > self.config["search"]["target_rt_tolerance"]:
+            self.log_precursor_df(precursor_df)
+
             self.reporter.log_string(
-                f"❌ {'rt_error':<15}: {self.com.rt_error:.4f} > {self.config['search']['target_rt_tolerance']}",
-                verbosity="info",
+                f"=== Classifier has been trained for {self.fdr_manager.current_version + 1} iteration(s); minimum number is {self.config['calibration']['optimization_lock_min_steps']} ===",
+                verbosity="progress",
+            )
+
+            if (
+                precursors_01FDR
+                > self.config["calibration"]["optimization_lock_target"]
+                and self.fdr_manager.current_version
```

Comment:
It's called version because its primary purpose is to allow for retrieval of a previously-trained classifier when we backtrack to the optimum value of a parameter after the optimizer has converged. I think it's an intuitive name, but please do explain why it bugs you; I am sure it can probably be improved. 

I agree, however, that it's a confusing thing to use in this context because it is just acting as a synonym of current_step, which would be more intuitive to call. I have changed this line so that it uses current_step instead.